Tools/Scripts/import-source-map-tests

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2024 Igalia S.L.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# 1. Redistributions of source code must retain the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer.
+# 2. Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials
+#    provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+# OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+# TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+
+# Updates the source map spec tests in LayoutTests/tc39-tg4/source-map-tests
+# with the latest upstream tests.
+
+import argparse
+import pathlib
+import re
+import shutil
+import sys
+
+from webkitpy.common.host import Host
+from webkitpy.common.system.filesystem import FileSystem
+from webkitpy.common.webkit_finder import WebKitFinder
+from webkitpy.common.checkout.scm.git import Git
+
+host = Host()
+host.initialize_scm()
+port = host.port_factory.get()
+layout_tests_directory = port.layout_tests_dir()
+source_map_repo_url = "https://github.com/tc39/source-map-tests.git"
+source_map_repo_branch = "main"
+
+def generate_readme_contents(git_hash):
+    return """\
+This directory contains the imported TG4 source map tests
+
+  - URL: {url}
+  - Commit: {git_hash}
+
+To update the tests, run the script at Tools/Scripts/import-source-map-tests
+with no arguments. This will checkout out the repo above and copy the new
+tests over. You can also specify a different repo and branch (see usage
+info with the -h flag).
+""".format(url=source_map_repo_url, git_hash=git_hash)
+
+def checkout_tests(continuation):
+    with host.filesystem.mkdtemp() as temporary_directory:
+        checkout_folder = "source-map-tests"
+        checkout_directory = host.filesystem.join(temporary_directory, checkout_folder)
+        print("Checking out tests from " + source_map_repo_url + " into " + checkout_directory)
+        Git.clone(source_map_repo_url, checkout_directory, host.executive)
```

Comment:
We probably want to make it possible for people to import from a local directory without cloning, too.